### ISSUE TITLE
Enable torch.max

### ISSUE THREAD

**Issue by imaihal** [torch-spyre/torch-spyre#453]
### What feature would you like to be added?

Run [torch.max](https://docs.pytorch.org/docs/stable/generated/torch.max.html) on spyre.

### Why is this feature needed?

This op is included important models.

**Comment by Sunny-Anand** [comment]
@imaihal  is this issue blocked? Can this be assigned to me for torchops squad work?

**Comment by imaihal** [comment]
@Sunny-Anand Below is the current implementation status of torch.max. There are still some remaining TODO items. I was planning to work them later, but feel free to work on them if you’d like.

---
I created a PR https://github.com/torch-spyre/torch-spyre/pull/657 to enable running torch.max.
The remaining TODOs are (as described in the PR):

- Support torch.ops.aten.max.default
This is torch.max() without the dim attribute, where the output is a scalar tensor. There was an issue about moving scalar tensors from Spyre to CPU (issue #1172), but it has already been closed, so we can try testing this again.

- Support the indices in the second output on Spyre
Currently, the indices are computed using torch.amax, but this runs on the CPU (CPU fallback). To run it on Spyre, we need to support torch.amax on Spyre. Another option is to use topk with k=1, since we can obtain indices using topk.

**Comment by tdoublep** [comment]
Any update on this? It is blocking us implementing online softmax in vLLM 

cc @ani300
